File: Harvard-CS50-Introduction_to_Computer_Science/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height, row, column, spaces;

    do
    {
        height = get_int("Height:");
    }
    while (height < 1 || height > 8);


//for rows
    for (row = 0 ; row < height ; row++)
    {
//for spaces
        for (spaces = 0 ; spaces < height - row - 1 ; spaces++)
        {
            printf(" ");
        }

//for columns
        for (column = 0 ; column <= row ; column++)
        {
            printf("#");
        }

        printf("\n");
    }

}

File: Harvard-CS50-Introduction_to_Computer_Science/sentimental-mario-more/mario.py
# TODO

from cs50 import get_int

while True:  # there is no do-while loop in python
    height = get_int("Please select height between 1 and 8: ")
    if (height <= 8 and height >= 1):
        break

row = 1

for i in range(0, height, 1):  # (starting point, end, incrementation)
    for j in range(0, height, 1):
        if i+j < height - 1:
            print(" ", end="")
        if i+j >= height - 1:
            print("#", end="")
    if height == height:
        print("  ", end="")
    for k in range(0, row, 1):  # if you use row instead of height as the limit, the loop does not run through unnecessarily more times without printing
        if k <= row:
            print("#", end="")
    print()
    row = row + 1
